### ISSUE TITLE
Release Notes v2.2.2

### ISSUE THREAD

**Issue by sadatmalik** [talentbeyondboundaries/tbbtalentv2#888]
# New Features

## New way of running stats on saved searches
Temporarily there are two ways of running stats, and two buttons "Run stats" and "New run stats".

They should both produce the same results, but the new way should get over one of the
limitations of the old way - for example, reported by Sarah Walder where she could not run stats
on searches showing candidates in Lebanon or Jordan because there were too many of them
(greater than 32,000). The new way of running stats should avoid that limitation.

Once we are happy that the new way of running stats produces the same results as the old way,
we can remove the old way altogether.

## Automated Messages
- Issue #1102 will work on a good summary of all the posts and content that we can use for release notes

## Candidate Chats
Admins belonging to a source partner organisation can now manage all their active chats with candidates from the 'Candidate Chats' tab on the 'Jobs' screen. Candidate results are searchable and sortable by name and candidate number, and can be filtered to show only candidates whose chats have unread posts.

## Removed the ability of users to manually associate a job with a list
The "submission list" is the main list associated with a job. It is automatically created when the job is created.
Its use is unchanged.

However, it used to be possible for a user to manually associate a job with any list - and that list would behave a bit like a submission list even though it wasn't. For example, adding a candidate to that list would automatically create a candidate opportunity. This ability to associate a job with any list wasn't widely used - but it did cause some nasty bugs with people accidentally creating or modifying candidate opportunities when they didn't mean to. So we decided it was best to remove the ability.

If you feel that you need that ability to manually associate jobs with lists, please let us know and perhaps we can find a different solution for you.

## General Improvements

- Update and sync relocating dependants to SF on the candidate case or on the candidate's visa check
- New UK visa check, containing the relocating dependants update & SF sync.
- Create a new job using a previous job as a template, this will copy over any job data that can be duplicated
- Upload a signed MOU document to a candidate case
- All fields now available to be used with 'Keyword Search' (which utilises elasticsearch functionality) in candidate search, which also increases the utility of its 'Base Search' functionality _(NB: I'll add a video for this one - SS)_
- Email validation improved and fully aligned across components and portals
- 'UNHCR Status' added as filter in candidate search
- Translation added to Admin Portal for improved code-sharing with Candidate Portal and to facilitate future non-English TC admin
- The "read only" for any user has been reviewed and now should work well for any kind of user. Read only users can create and modify their own lists and searches - but they can't change anything else.
- Salesforce errors appearing on admin portal to confirm if a Salesforce update has or hasn't been successful
- Destination preferences moved from Mini Intake and added to the candidate's registration process on the candidate portal. On the admin portal this data appears and is editable under the General tab of the candidate's profile.
- When setting the 'Next Step' on a job or candidate opportunity, the audit information (date & user) is automatically added to the next step and no longer needs to be entered manually as was previously directed.
- Added 'IETLS Score' to default columns, replacing 'Updated Date'. 
- The updated date can be now viewed in the candidate profile card on the General tab under Additional Information.

## Data Improvements
- Salesforce Candidate Opportunity relocating dependants stats now auto-updated at 'Relocated' stage
- Support added for new Salesforce 'MOU' and 'Training' stages
- Elasticsearch index updated in all data interactions with candidates
- Candidates' managing partner can now be reassigned in bulk from Settings > System Admin API, improving efficiency and data integrity
- Consolidate UNHCR Registered and UNHCR Status fields, now stored in a single UNHCR Status field which provides most informative and up to date data.
- Handle invisible LSEP characters in some candidates' job experience descriptions that was preventing CV downloads

# UI / UX Enhancements
- Preview key intake information in the intake headers to allowing easy access when intake panels are closed.
- Larger more mobile friendly tab navigation on candidate portal profile
- Job Chat link functionality replaced with bespoke version to include improved menus, preview display of linked-to content and URL validation
- Custom partner logos for Map India and IRC

## Other UI / UX Enhancements
- Fix overlapping navbar when scrolling on candidate portal
- Better alignment of content under tabs in candidate portal
- AU visa check has updated style consistent with other visa checks
- Added subtle box shadow to buttons to provide extra dimension on admin portal
- Proper hover-over link behaviour added to Candidate Portal
- Tooltip added to help users in updating candidate lists
- Candidate search profile card restores previous scrollbar position for new candidate selections
- 'Copy list' and 'save selection to list' pop ups have improved layout
- Job Chat notification * now appears as a red dot consistent with notification UI on other apps
- Neatened the filter checkboxes for job and opportunity searches 
- Updated the mark as read button and removed if a chat is already read 
- Disabled buttons appear greyed out to help active buttons stand out on the page
- Candidate search card fills the window height
- Switch off Stats menu access for employer users
- Display TC splash image on reset password screens

# Performance Improvements

- Introduced Redis caching layer in AWS cloud environments to improve general system performance
- Faster search results for name, number, email, and phone lookups in quick search bar
- Introduced enhanced TC loggers for improved user action visibility in local and cloud logs which will help diagnose future performance (and other) issues
- Implemented mechanisms for Redis cache evictions to avoid stale data in the cache
- Implemented system API to manually trigger Redis cache evictions if required
- Detect high resource utilisation and auto-restart TC cloud services 
- Update database indexes in key tables to improve performance of job tabs
- Set max time to live values for cache entries to prevent cache pollution
- Transfer minimum required data between TC services and web clients for candidate name and number search; and for list and search browsing previews

# Security Fixes

- Added MFA to the AWS Production environment for an additional layer of security
- Rotated, deactivated, and removed unused and quarantined AWS access keys for security breach mitigation

# Bug Fixes
- Visa job checks can now only be created for the job destination's visa check
- Candidate fields no longer being overwritten by older cached candidate - this resolves candidate fields such as intake data, partner and status reverting back to an older cached value
- Weekly candidate syncing to Salesforce now fully operational
- Performance issues arising from looped processing of large result sets resolved using Entity Manager's clear() method
- Fixed translation display issues in Candidate Portal registration
- CVs now downloading successfully from the candidate search results dropdown menu and clickable icon
- Candidate Prospect chats no longer being unnecessarily auto-created on candidate profile view, instead created via a button in the tab with help text; viewable by all appropriate admins
- Search tips no longer erroneously opened on 'Enter' keydown in candidate search inputs
- Candidate search correctly handling default saved search for new TC users
- Validation tightened and informative error-handling added to Job publishing from Salesforce URL
- IELTS scores of 0 now displayed in candidate search results
- Help text clarified on Admin Portal date picker component
- Potential password exposure in server logs removed
- Pop ups with a 'select list' dropdown made larger to avoid list names being cut off
- Fixed a bug where the column select modal resulted in an error when closed with the Escape key

# Developer Notes

## Test Coverage

- Fixed Gatling performance test database connection issues
- Paused automatic execution of Gatling performance tests in cloud environments to avoid unnecessary system load

## Code Refactoring

- Deprecated candidate visa job check fields removed from code
- Emoji picker code refactored for better readability

## Continuous Integration & Deployment

- Elasticsearch starter scripts for new devs
- Updated terraform directory structure and git actions for better control over auto deployments
- Deactivated automatic builds in tc-build git action to avoid duplicate builds when pushing to staging and master branches

## Cloud Enhancements

- Deployed AWS Elasticache Redis clusters to reduce DB user lookups and improve general system performance
- Renewed RDS database certificates on AWS cloud production and staging environments

## Logging and Monitoring

- Enhanced log builders for improved local and cloud log diagnostics
- Fixed repetitive expired JWT token errors by properly handling the first occurrence in the front end; requiring a fresh user login in this scenario
- Deployed AWS Cloudwatch alerts and AWS Lambdas to auto-restart services on high resource utilisation detection

**Comment by samschlicht** [comment]
Pic for '[Partner] Candidate Chats' tab:

<img width="1465" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3lIQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bf7924d577657f1061acb31756bfdf48c8b25d03/Screenshot%202024-08-26%20at%2012.23.33%E2%80%AFPM.png" alt="Screenshot 2024-08-26 at 12.23.33 PM.png" />

**Comment by cazcam34** [comment]
Pic for new way of running stats button
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNXFpQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--905bc9c0310120552505f4a2e88146cd09fa6025/Screenshot%202024-09-02%20at%202.39.06%E2%80%AFPM.png" alt="Screenshot 2024-09-02 at 2.39.06 PM.png" />

**Comment by cazcam34** [comment]
Image for automated job chats?
<img width="1266" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN2VpQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--54a2199115a050d0d024d27678bdfab24c50c982/Screenshot%202024-09-02%20at%203.40.58%E2%80%AFPM.png" alt="Screenshot 2024-09-02 at 3.40.58 PM.png" />

**Comment by cazcam34** [comment]
Drafted release notes are here: https://talent-catalog.github.io/talentcatalog/release_notes_v2_2_2.html
I wonder if instead of the Run Stats button as a new feature, we could do a 1 pager highlighting all the places where we've put in 'speed fixes'. Eg. the candidate name/number search, the stats, the lists etc. Thoughts?

Need to:
- [x] Page explanation for automated posts
- [ ] Page explanation for candidate chats tab
- [ ] Page explanation for run stats button?
- [x] Better image for highlight intake data (candidate test number 96406)
- [x] Better image for new way for running stats (unless we change to a 'speed improvements' page)

**Comment by cazcam34** [comment]
<img width="1044" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBLytxQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5546a75f0b0e0de05167052dc50839dbb868792a/IntakeSummary.png" alt="IntakeSummary.png" />

**Comment by cazcam34** [comment]
Q: What do we do for 'New Features' that don't require a full page of notes, but do have more than a bullet point. Eg. All of the new features in these release notes minus Automated Posts (which can have it's own page). I have created a new candidate card which extends the full width of the page as it gives more space for text, and it also works well for features that don't have a suitable image (e.g. Removed the ability of users to manually associate a job with a list
feature). Be good to get teams input in our meeting Tuesday.
